src/GraphResponseHandler.ts

```diff
@@ -49,20 +49,24 @@ export class GraphResponseHandler {
 	 * @returns A promise that resolves to a document content
 	 */
 	private static parseDocumentResponse(rawResponse: Response, type: DocumentType): Promise<any> {
-		if (typeof DOMParser !== "undefined") {
-			return new Promise((resolve, reject) => {
-				rawResponse.text().then((xmlString) => {
-					try {
-						const parser = new DOMParser();
-						const xmlDoc = parser.parseFromString(xmlString, type);
-						resolve(xmlDoc);
-					} catch (error) {
-						reject(error);
-					}
+		try {
+			if (typeof DOMParser !== "undefined") {
+				return new Promise((resolve, reject) => {
+					rawResponse.text().then((xmlString) => {
+						try {
+							const parser = new DOMParser();
+							const xmlDoc = parser.parseFromString(xmlString, type);
+							resolve(xmlDoc);
+						} catch (error) {
+							reject(error);
+						}
+					});
 				});
-			});
-		} else {
-			return Promise.resolve(rawResponse.body);
+			} else {
+				return Promise.resolve(rawResponse.body);
+			}
+		} catch (error) {
```

Comment:
We can omit try-catch-throw since we aren't inspecting the error object. Did you intend to inspect this error? same with [R130](https://github.com/microsoftgraph/msgraph-sdk-javascript/pull/144/files#diff-097803c7ff7847cd97efb185745e148dR130)

Reply:
Yes, with line number 130 I did inspect the error object. I have added this for the code clarity to bascially understand that this will throw error.